source/go-perun/app_tutorial/app.rst

```diff
@@ -0,0 +1,712 @@
+Finally, we convert the bytes to their respective field values by calling ``makeFieldValueArray``.
+
+.. code-block:: go
+
+    // DecodeData decodes the channel data.
+    func (a *TicTacToeApp) DecodeData(r io.Reader) (channel.Data, error) {
+        d := TicTacToeAppData{}
+
+        var err error
+        d.NextActor, err = readUInt8(r)
+        if err != nil {
+            return nil, errors.WithMessage(err, "reading actor")
+        }
+
+        grid, err := readUInt8Array(r, len(d.Grid))
+        if err != nil {
+            return nil, errors.WithMessage(err, "reading grid")
+        }
+        copy(d.Grid[:], makeFieldValueArray(grid))
+        return &d, nil
+    }
+
+Validate Initial State
+~~~~~~~~~~~~~~~~~~~~~~
+We will need to identify if an initial state is valid or not.
+Hence, we create ``ValidInit`` that takes the channel's parameters ``channel.Params`` and a channel state ``channel.State`` as arguments.
+If the given arguments do not match the expected ones for a valid app channel initialization, it should return an error.
+
+**Channel participants.**
+Check if the number of actual channel participants matches the expected participants.
+
+.. code-block:: go
+
+    // ValidInit checks that the initial state is valid.
+    func (a *TicTacToeApp) ValidInit(p *channel.Params, s *channel.State) error {
+        if len(p.Parts) != numParts {
+            return fmt.Errorf("invalid number of participants: expected %d, got %d", numParts, len(p.Parts))
+        }
+
+**Game data.**
+Validate the data type of the ``channel.State``'s data. It must be of type ``TicTacToeAppData``.
+
+.. code-block:: go
+
+        appData, ok := s.Data.(*TicTacToeAppData)
+        if !ok {
+            return fmt.Errorf("invalid data type: %T", s.Data)
+        }
+
+**Grid.**
+The grid must be empty. This is easily checkable by creating a new instance of ``TicTacToeAppData`` and comparing its gird with the given one.
+
+.. code-block:: go
+
+        zero := TicTacToeAppData{}
+        if appData.Grid != zero.Grid {
+            return fmt.Errorf("invalid starting grid: %v", appData.Grid)
+        }
+
+**Finalization.**
+Verify that the channel state is not finalized.
+Remember that a final state cannot be further progressed. Therefore it would not make sense to accept one here.
+
+.. code-block:: go
+
+        if s.IsFinal {
+            return fmt.Errorf("must not be final")
+        }
+
+**Actor index.**
+Validate the index of the ``NextActor``
+If no deviations are found, ``nil`` is returned.
+
+.. code-block:: go
+
+        if appData.NextActor >= numParts {
+            return fmt.Errorf("invalid next actor: got %d, expected < %d", appData.NextActor, numParts)
+        }
+        return nil
+    }
+
+.. _app_validate_transition:
+
+Validate Transitions
+~~~~~~~~~~~~~~~~~~~~
+``ValidTransition`` is a critical part of the app implementation in which we check if a given game move is valid or not.
```

Comment:
This is only the off-chain valid transition function. Should be pointed out. Even more critical is actually the on-chain validTransition function, because it ultimatively decides what we can enforce.